src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -696,22 +717,23 @@ private Compilation CreateEmptyCompilation()
             /// Add all appropriate references to the compilation and set it as our final compilation
             /// state.
             /// </summary>
-            /// <param name="generatorInfo">The generator info that contains the last run of the documents, if any exists, as
-            /// well as the driver that can be used to run if need to.</param>
-            /// <param name="compilationWithStaleGeneratedTrees">The compilation from a prior run that contains generated trees, which
-            /// match the states included in <paramref name="generatorInfo"/>. If a generator run here produces
-            /// the same set of generated documents as are in <paramref name="generatorInfo"/>, and we don't need to make any other
-            /// changes to references, we can then use this compilation instead of re-adding source generated files again to the
-            /// <paramref name="compilationWithoutGeneratedDocuments"/>.</param>
+            /// <param name="compilationWithStaleGeneratedTrees">The compilation from a prior run that contains
+            /// generated trees, which match the states included in <paramref name="allSyntaxTreesParsedState"/>'s
+            /// generator info. If a generator run here produces the same set of generated documents as are in <paramref
+            /// name="allSyntaxTreesParsedState"/>'s generator info, and we don't need to make any other changes to
+            /// references, we can then use this compilation instead of re-adding source generated files again to the
+            /// compilation that <paramref name="allSyntaxTreesParsedState"/> points to.</param>
             private async Task<FinalState> FinalizeCompilationAsync(
                 SolutionCompilationState compilationState,
-                Compilation compilationWithoutGeneratedDocuments,
-                CompilationTrackerGeneratorInfo generatorInfo,
+                AllSyntaxTreesParsedState allSyntaxTreesParsedState,
                 Compilation? compilationWithStaleGeneratedTrees,
```

Comment:
Didn't you add the w/ stale generated trees to the AllSyntaxTreesParsedState? So we could drop this too?

Reply:
yes'ish. that's an open question.  technically it might change things slightly.  so i've held off on that change until i can guarantee it either has the same semantics *or* that it is either nuetral or makes things better.  i'm slightly concerned that it might break something, so it's going to be held off till a future pr.